### DISCUSSION TITLE
How to regexp match hostname in external_relay plugin?

### DISCUSSION THREAD

**Discussion by lenar** [rspamd/rspamd#5145] [Q&A]
What I want to do is explained by this (external_relay.conf contents):

```
enabled = true;
rules {
  EXTERNAL_RELAY_AUTHENTICATED {
    strategy = "hostname_map";
    hostname_map = ["/\.outlook\.com$/"];
  }
}
```

Does not seem to work. I don't see any other way to skip checking SPF against wrong servers when the relaying is done by Exchange Online (multiple (3) Received: headers after the actual sending server, but all them have *.outlook.com hostname).

**Comment by fatalbanana** [comment]
I would suggest to live with the SPF failures, they are perfectly normal.

The way to use regexp is with map augmentations, e.g.

```
hostname_map = "regexp;/path/to/map";
```
